### DISCUSSION TITLE
Add more alternative representations of surface-partitioning for organics

### DISCUSSION THREAD

**Discussion by claresinger** [open-atmos/PySDM#632] [Ideas]
The current compressed film framework implemented is one (very simple) option based on [Lowe et al (2019)](http://www.nature.com/articles/s41467-019-12982-0) and [Ovadnevaite et al (2017)](http://www.nature.com/articles/nature22806). We'd like to 1) rename this option and 2) add another (slightly more complex version) based on [Ruehl et al (2016)](https://www.sciencemag.org/lookup/doi/10.1126/science.aad4889).

I have preliminary python code for Ruehl. This model iteratively solves for the fraction of organic that stays at the surface in the film (vs. dissolves into the bulk phase), and thus adds one more degree of freedom to the representation of the surface-partitioning. 

```
def sigma_Ruehl(T, v_wet, v_dry, f_org, nu_org):
    Cb_iso = (f_org*v_dry/nu_org) / (v_wet/nu_w) # = C_bulk / (1-f_surf)
    r_wet = ((3 * v_wet) / (4 * np.pi))**(1/3) # m - wet radius
    A_iso = (4*np.pi*r_wet**2) / (f_org*v_dry*NA/nu_org) # m^2 = A*f_surf

    f = lambda f_surf: Cb_iso*(1-f_surf) - C0*np.exp(((A0**2 - (A_iso/f_surf)**2)*m_sigma*NA)/(2*R*T))
    sol = root_scalar(f, bracket=[0,1])
    f_surf = sol.root
    
    sgm = sgm_w - (A0 - A_iso/f_surf)*m_sigma # m^2 * J/m^2 = J = N*m --> N/m - N*m ?
    sgm = np.minimum(np.maximum(sgm, sgm_min), sgm_w)
    return sgm
```

**Comment by slayoo** [comment]
@claresinger is ``nu_org`` needed to be added as particle attribute? If so, how to initialize and vary it?

**Comment by claresinger** [comment]
Yes... this may require a longer discussion @rxward. 

In this model of the surface-partitioning we explicitly need to know the "molar volume" of the organic phase in the surface tension calculation. It's similar to how we needed the molar volume [previously](https://github.com/atmos-cloud-sim-uj/PySDM-examples/blob/5080d24b000239386f4b17d69bb77a7e2dd8364d/PySDM_examples/Lowe_et_al_2019/aerosol.py#L56-L75) in the calculation of Köhler curves, but we were able to roll this into `kappa`. I'm not sure if something similar can be done here, I need to think about it more, but I will guess maybe it can't and we'll need to add another attribute?

**Comment by slayoo** [comment]
@claresinger @rxward any news here?

**Reply by claresinger** [reply]
I think @rxward and I are agreed that to implement this Ruehl implementation we will need either:
1) A constant to describe the organic composition (if we have only one species then this is constant).
2) Another superdroplet attribute for the organic composition because if particles are colliding then `nu_org` can change.

What we are interested in testing with PySDM is just looking at one species at a time because this is what Ryan has been doing in the lab. That is probably also easier to implement, so maybe we start with that? We will want to be able though to specify `nu_org` from the `settings.py` script in the examples, so it shouldn't be hardcoded in the `constants.py` file.

**Comment by slayoo** [comment]
Re setting a constant within examples, we can do it as with the sgm_org and delta_min, see here:
https://github.com/atmos-cloud-sim-uj/PySDM/blob/master/PySDM/physics/surface_tension/compressed_film.py

**Reply by claresinger** [reply]
Yes! Let's do it like that for now. I will open a PR tomorrow.

**Reply by claresinger** [reply]
Actually, looking at that, I suppose it's the same idea. If we wanted to have multiple organic species then we'd probably need `sgm_org` to be species-dependent. But as far as I know, @rxward and I have no plans to look at multiple species at all or at least for a long while.

**Reply by rxward** [reply]
Yes, initially we'll only need to run isolated species!
